Add set_interface for access by GDNative

BastiaanOlij opened the pull request:
Ok, I've completely redone this and added a few additional changes to this.

We can now call the set_interface through a new GDNative ARVR API call. 

As I'm adding a new API level I've also added another often requested feature and that is to get access to the depth buffer. 
This is important in XR for re-projection during frame drops and to enable stuff like LIV (https://www.liv.tv/) and other mixed reality solutions. 

I've also added an option to provide an external depth buffer alongside the external color buffer. 

The depth buffer changes are pretty experimental especially on GLES2 as there is no way to switch between render buffer and texture.

akien-mga commented:
Still crashing with:
```
modules/gdnative/arvr/arvr_interface_gdnative.cpp:84:20: runtime error: member access within misaligned address 0x00000000000a for type 'const struct godot_arvr_interface_gdnative', which requires 8 byte alignment
0x00000000000a: note: pointer points here
<memory cannot be printed>
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] bin/godot.x11.tools.64s() [0x171b67a] (/home/runner/work/godot/godot/platform/x11/crash_handler_x11.cpp:54)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f053eab4210] (??:0)
[3] ARVRInterfaceGDNative::set_interface(godot_arvr_interface_gdnative const*) (/home/runner/work/godot/godot/modules/gdnative/arvr/arvr_interface_gdnative.cpp:84)
[4] ARVRInterfaceGDNative::_set_interface(long) (/home/runner/work/godot/godot/modules/gdnative/arvr/arvr_interface_gdnative.cpp:90)
```

For the reference, here's what that CI Linux build with sanitizers and test project does:

https://github.com/godotengine/godot/blob/6b48fce2b54af26cc8e640129d8b93823f53226f/.github/workflows/linux_builds.yml#L188-L210

BastiaanOlij commented:
Note before this gets lost and forgotten. After discussion on godot chat about this we've decided to do this more proper and add the required function to the ARVR GDNative API. Will find some time for that soonish.

BastiaanOlij committed:
ARVR GDNative enhancements:
- add set_interface function
- add access to depth buffer
- add supplying a depth buffer from an ARVR plugin

BastiaanOlij commented:
Ok, so I redid this whole thing and just updated the opening text. Also added access to the depth buffer.

akien-mga commented:
Thanks!
